module/builder/consensus/builder.go

```diff
@@ -153,8 +153,8 @@ func (b *Builder) BuildOn(parentID flow.Identifier, setter func(*flow.Header) er
 // 2) traverse forward all unfinalized(pending) blocks starting from last finalized block.
 // For each block that is being traversed we will collect execution results and add them to execution tree.
 func (b *Builder) repopulateExecutionTree() error {
-	finalizedSnapshot := b.state.Final()
-	finalized, err := finalizedSnapshot.Head()
+	finalizedHeader := b.state.Final()
```

Comment:
was this intentional ? the `state.Final` is still a `snapshot`.

Reply:
Oops, will change this